src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,232 @@
+                        _ => Task.CompletedTask
+                    };
+            }
+
+            private static bool IsArgument(IOperation? operation, [NotNullWhen(returnValue: true)] out IArgumentOperation? argumentOperation)
+            {
+                while (operation is not null)
+                {
+                    if (operation is IArgumentOperation tmpArgumentOperation)
+                    {
+                        argumentOperation = tmpArgumentOperation;
+                        return true;
+                    }
+
+                    operation = operation.Parent;
+                }
+
+                argumentOperation = null;
+                return false;
+            }
+
+            private static bool IsReturn(IOperation? operation)
+            {
+                while (operation is not null)
+                {
+                    if (operation is IReturnOperation)
+                    {
+                        return true;
+                    }
+
+                    operation = operation.Parent;
+                }
+
+                return false;
+            }
+
+            private async Task VisitLiteralAsync(ILiteralOperation literalOperation, CancellationToken cancellationToken)
+            {
+                if (literalOperation.Type is null)
+                {
+                    return;
+                }
+
+                await AddOperationAsync(literalOperation, literalOperation.Type, cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task VisitReturnAsync(IReturnOperation returnOperation, CancellationToken cancellationToken)
+            {
+                if (returnOperation.ReturnedValue is null)
+                {
+                    return;
+                }
+
+                await VisitAsync(returnOperation.ReturnedValue, cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task AddOperationAsync(IOperation operation, ISymbol symbol, CancellationToken cancellationToken)
+            {
+                _ = await ProgressCollector.TryReportAsync(
+                        Solution,
+                        operation.Syntax.GetLocation(),
+                        symbol,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task TrackArgumentsAsync(ImmutableArray<IArgumentOperation> argumentOperations, CancellationToken cancellationToken)
+            {
+                var collectorsAndArgumentMap = argumentOperations
+                    .Where(ShouldTrackArgument)
+                    .Select(argument => (collector: Clone(), argument))
+                    .ToImmutableArray();
+
+                var tasks = collectorsAndArgumentMap
+                    .Select(pair => pair.collector.VisitAsync(pair.argument, cancellationToken));
+
+                await Task.WhenAll(tasks).ConfigureAwait(false);
```

Comment:
this will not really be concurrent.  you'd need `.Select(pair => Task.Run(...)`. for that.